tests/jballerina-unit-test/src/test/resources/test-src/expressions/rawtemplate/raw_template_literal_test.bal

```diff
@@ -0,0 +1,230 @@
+
+function testSubtyping1() {
+    int x = 10;
+    int y = 20;
+
+    Template1 t = `${x} + ${y} = ${x + y}`;
+    assert("10 + 20 = 30", t.toString());
+}
+
+type Template2 object {
+    public string[] strings = [];
+    public [int, string, float] insertions = [];
+};
+
+function testSubtyping2() {
+    int x = 25;
+    string s = "foo";
+    float f = 12.34;
+
+    Template2 t = `Using tuples: ${x}, ${s}, ${f}`;
+    assert("Using tuples:  ,  , ", t.strings.toString());
+    assert(x, t.insertions[0]);
+    assert(s, t.insertions[1]);
+    assert(f, t.insertions[2]);
+
+    object {
+        public string[] strings = [];
+        public [int, string, anydata...] insertions;
+        string name = "";
+
+        function __init() {
+            self.insertions = [];
+        }
+    } temp2 = `Using tuples 2: ${x}, ${s}, ${f}`;
+
+    assert("Using tuples 2:  ,  , ", temp2.strings.toString());
+    assert(x, temp2.insertions[0]);
+    assert(s, temp2.insertions[1]);
+    assert(f, temp2.insertions[2]);
+
+    object {
+        public string[] strings = [];
+        public [anydata...] insertions = [];
+        string name = "";
+    } temp3 = `Using tuples 3: ${x}, ${s}, ${f}`;
+
+    assert("Using tuples 3:  ,  , ", temp3.strings.toString());
+    assert(x, temp2.insertions[0]);
+    assert(s, temp3.insertions[1]);
+    assert(f, temp3.insertions[2]);
+}
+
+const FOO = "Foo";
+const BAR = "Bar";
+
+type FooBar FOO|BAR;
+
+function testSubtyping3() {
+    int x = 10;
+
+    object {
+        public FooBar[] strings = [];
+        public int[] insertions = [];
+    } temp1 = `Foo${x}Bar`;
+
+    assert(<string[]>["Foo", "Bar"], temp1.strings);
+    assert(10, temp1.insertions[0]);
+}
+
+function testUsageWithQueryExpressions() {
+    int[] arr = [100, 200, 300, 400, 500];
+
+    ob:RawTemplate[] queries = from var x in arr
+        select `INSERT INTO product ('id', 'name', 'price') VALUES (${x}, ${x + 10}, ${x + 20})`;
+
+    foreach var x in queries {
+        assert("INSERT INTO product ('id', 'name', 'price') VALUES ( ,  ,  )", x.strings.toString());
+    }
+
+    int i = 0;
+    foreach var x in queries {
+        assert(string `${arr[i]} ${arr[i] + 10} ${arr[i] + 20}`, x.insertions.toString());
+        i += 1;
+    }
+}
+
+public type Value ()|int|float|decimal|string|xml;
+
+public type ParameterizedQuery object {
+    public string[] strings = [];
+    public Value[] insertions = [];
+};
+
+function testUsageWithQueryExpressions2() {
+    var data = [
+        {name: "Alice1"},
+        {name: "Alice2"},
+        {name: "Alice3"}
+    ];
+
+    ParameterizedQuery[] queries = from var rec in data select `INSERT INTO People (name) values (${rec.name});`;
+
+    foreach var x in queries {
+        assert("INSERT INTO People (name) values ( );", x.strings.toString());
+    }
+
+    int i = 0;
+    foreach var x in queries {
+        assert(data[i]["name"], x.insertions.toString());
+        i += 1;
+    }
+}
+
+function testUseWithVar() {
```

Comment:
This is currently failing right.

Reply:
Yes, I haven't pushed the updated tests yet. Will push them once the negative tests are also updated

Reply:
Pushed the updated tests